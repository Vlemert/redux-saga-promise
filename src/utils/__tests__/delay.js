// @flow

import delay from '../delay';

jest.useFakeTimers();

describe('delay', () => {
  test('returns a promise that resolves after a specific time', async () => {
    const resolvedTimers = [];

    const setDelay = (time: number) => delay(time).then(() => resolvedTimers.push(time));

    setDelay(1000);
    setDelay(2000);
    setDelay(3000);
    setDelay(3001);
    setDelay(3001);

    const moveTime = (time: number) => {
      jest.runTimersToTime(time);

      return new Promise(resolve => setImmediate(resolve));
    };

    await moveTime(1000);
    expect(resolvedTimers).toMatchSnapshot();

    await moveTime(1000);
    expect(resolvedTimers).toMatchSnapshot();

    await moveTime(500);
    expect(resolvedTimers).toMatchSnapshot();

    await moveTime(499);
    expect(resolvedTimers).toMatchSnapshot();

    await moveTime(1);
    expect(resolvedTimers).toMatchSnapshot();

    await moveTime(1);
    expect(resolvedTimers).toMatchSnapshot();
  });
});
