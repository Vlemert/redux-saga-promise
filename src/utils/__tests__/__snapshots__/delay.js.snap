// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delay returns a promise that resolves after a specific time 1`] = `
Array [
  1000,
]
`;

exports[`delay returns a promise that resolves after a specific time 2`] = `
Array [
  1000,
  2000,
]
`;

exports[`delay returns a promise that resolves after a specific time 3`] = `
Array [
  1000,
  2000,
]
`;

exports[`delay returns a promise that resolves after a specific time 4`] = `
Array [
  1000,
  2000,
]
`;

exports[`delay returns a promise that resolves after a specific time 5`] = `
Array [
  1000,
  2000,
  3000,
]
`;

exports[`delay returns a promise that resolves after a specific time 6`] = `
Array [
  1000,
  2000,
  3000,
  3001,
  3001,
]
`;
